src/NServiceBus.AcceptanceTests/Routing/When_extending_command_routing.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_extending_command_routing : NServiceBusAcceptanceTest
+    {
+        static string ReceiverEndpoint => Conventions.EndpointNamingConvention(typeof(Receiver));
+
+        [Test]
+        public async Task Should_route_commands_correctly()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<Sender>(b =>
+                    b.When(c => c.EndpointsStarted, async session =>
+                    {
+                        await session.Send(new MyCommand());
+                        await session.Send(new MyCommand());
+                        await session.Send(new MyCommand());
+                        await session.Send(new MyCommand());
+                    })
+                )
+                .WithEndpoint<Receiver>()
+                .Done(c => c.MessageDelivered >= 4)
+                .Repeat(r => r.For(Transports.Default))
```

Comment:
in that case we don't need a repeat and can move the assert outside the Scenario.Define().Run(); or am I missing something?